### ISSUE TITLE
[Bug]: SD1.5 v-pred checkpoints don't work properly after the latest update

### ISSUE THREAD

**Issue by zxlostsoul** [lllyasviel/stable-diffusion-webui-forge#1109]
Some of the latest changes are broken 1.5 v-pred models.

Model example: https://civitai.com/models/199059/9th-tail

It has .yaml config with
```
  params:
    parameterization: "v"
```

But for some reason it doesn't work like before.

Tested on clean webui-forge setup.

It creates mess instead of images:
![00017-771476832](https://github.com/user-attachments/assets/86d1f66a-3a62-4f8e-bf4b-d44a5ef7728c)

**Comment by rabidcopy** [comment]
It seems things relating to custom configurations for checkpoints got thrown out. Don't see any apparent way to set prediction types in the current repo state. https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/bccf9fb23a2cbc9d42091a856b3d35b2c4414d51#diff-b710a9b8e9fbcc5bc5a014f938c9c74564c1dcfc86929f0dc9ff643ba3fe7873

**Comment by zxlostsoul** [comment]
For those who need v-pred models right now:

Switch git repo to commit e3522c89191a01e0dd5855abbfd15cb685be3634 , it's the last commit where checkpoint configs works (02.08.2024).

Commit bc9977a305391ec389e327f17df0bc33dd0ad472 is breaking it.

Hope there will be other way to set prediction type soon...

**Comment by alesimula** [comment]
@lllyasviel Any update on this? A lot of models just stopped working alltogether

**Comment by geriberton** [comment]
I also could not load a v-prediction model correctly, it seems to be ignoring the lines in the yaml and produces burnt images because of it.

**Comment by rabidcopy** [comment]
I've made a patch that allows you to override the prediction type used for sampling under Sampler parameters.
![image](https://github.com/user-attachments/assets/0ceb356a-4d37-463d-87e6-1ce9cf578d05)
Needs to be applied prior to the model being loaded for inference. (Restart or load another model if it's still stuck in eps.)
Save the following txt file in the root of your Forge folder and run `git apply prediction.txt`.
[prediction.txt](https://github.com/user-attachments/files/17334045/prediction.txt)

**Comment by DenOfEquity** [comment]
Nice work, rabidcopy. If the prediction_type is set at the start of sampling instead, there's no issue with needing to apply before model is loaded. [branch](https://github.com/DenOfEquity/stable-diffusion-webui-forge/tree/vprediction-hacky). *Settings > User interface > Quicksettings list* can be used to add your new setting to the UI. I'm unsure about merging though, it feels just a bit too much like a hack.
Then I realised it could be done entirely with an extension, so added it to [this](https://github.com/DenOfEquity/forgeFlux_dualPrompt).

**Comment by rabidcopy** [comment]
> Then I realised it could be done entirely with an extension, so added it to [this](https://github.com/DenOfEquity/forgeFlux_dualPrompt).

Nice, I will admit the former is indeed hacky.

**Comment by somelinuxguy** [comment]
I would love to see this repaired, but getting the feeling this is a low priority and we should just stick with SDXL and Flux models.

**Comment by catboxanon** [comment]
@DenOfEquity Sorry to bother, but could you re-consider merging that patch, and furthermore looking into https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/545 since it's closely related for v-pred models? It's preventing new foundational models like this one from being used, and also factoring into their decision on whether to continue training them or not due to lack of UI support (despite their clear benefits). https://civitai.com/models/833294?modelVersionId=962003

**Comment by alesimula** [comment]
@catboxanon 

I don't get it, tried the latest version with the model [EasyFluff_v10-Prerelease](https://huggingface.co/zatochu/EasyFluff/tree/main) with the yaml file in the same directory as the model, it's still not generating properly, am I missing something?

![image](https://github.com/user-attachments/assets/ccdd251d-b3f3-4dde-a072-d29537fb5498)

**Comment by catboxanon** [comment]
Re-opening so the previous .yaml file loading support can be looked into. Or perhaps another method to fix it.

**Comment by rabidcopy** [comment]
I can only suggest my patch from several comments above to set the prediction for 1.5 models. I don't plan to make a PR but anyone else is free to adapt it while not causing unforeseen bugs or conflicts with other related code. (The latter is out of scope for me.)

**Comment by catboxanon** [comment]
Should be fixed now. https://github.com/lllyasviel/stable-diffusion-webui-forge/pull/2224

**Comment by alesimula** [comment]
> Should be fixed now. #2224

I have tried the same model I have linked in my previous comment, nothing seems to have changed, have you tested it and is it working for you?

![image](https://github.com/user-attachments/assets/2b64c036-1cf2-4635-ab17-cc16f2677804)

**Comment by catboxanon** [comment]
Sorry, that's my bad, missed a pretty obvious typo. I tested with that model and config to be sure and it works for me now. https://github.com/lllyasviel/stable-diffusion-webui-forge/pull/2228

**Comment by alesimula** [comment]
> Sorry, that's my bad, missed a pretty obvious typo. I tested with that model and config to be sure and it works for me now. (#2228)

Thanks, I can confirm it's working now

**Comment by DiegoRRR** [comment]
> Save the following txt file in the root of your Forge folder and run `git apply prediction.txt`. [prediction.txt](https://github.com/user-attachments/files/17334045/prediction.txt)
I did. (but had to put it in webui folder instead of root for it to find the folders) I restarted Forge and Firefox. I went to Settings > Sampler parameters and set "Overwrite prediction type" to "v_prediction", applied the settings. But it still generates mosaic. What did I forget to do?
![20241214225738-2650974005-toon_illustrious__susamix4Noobai_v4NoobaixlVPred075s](https://github.com/user-attachments/assets/8636ec53-6ccf-4678-b318-6cd6b562333b)
